lua/orgmode/export/init.lua

```diff
@@ -172,4 +47,163 @@ function Export._exporter(cmd, target, on_success, on_error)
+  local target = vim.fn.fnamemodify(file, ':p:r') .. '.' .. opts.extension
+  local emacs = config.emacs_config.executable_path
+  local emacs_config_path = config.emacs_config.config_path
+  if vim.fn.executable(emacs) ~= 1 then
+    return utils.echo_error('emacs executable not found. Make sure emacs is in $PATH.')
+  end
+
+  local cmd = {
+    emacs,
+    '-nw',
+    '--batch',
+    '--load',
+    emacs_config_path,
+    string.format('--visit=%s', file),
+    string.format('--funcall=%s', opts.command),
+  }
+
+  return Export._exporter(cmd, target, nil, function(err)
+    table.insert(err, '')
+    table.insert(err, 'NOTE: Emacs export issues are most likely caused by bad or missing emacs configuration.')
+    return utils.echo_error(string.format('Export error:\n%s', table.concat(err, '\n')))
+  end)
+end
+
+Export.emacs_beamer = Export.emacs
+
+function Export.prompt()
+  local keys = {
+    emacs = 'e',
+    emacs_beamer = 'b',
+    pandoc = 'p',
+  }
+
+  local submenu = function(key, label, extension, exporters)
+    local commands = {
+      {
+        label = 'quit',
+        key = 'q',
+      },
+    }
+
+    local exporters_names = {}
+
+    for name, opts in pairs(exporters) do
+      table.insert(exporters_names, name)
+
+      opts.extension = extension
+
+      local exporter_label = name
+      if opts.command then
+        exporter_label = string.format('%s (%s)', name, opts.command)
+      else
+        exporter_label = name
+      end
+
+      table.insert(commands, {
+        label = exporter_label,
+        key = keys[name],
+        action = function()
+          return Export[name](opts)
+        end,
+      })
+    end
+
+    table.sort(commands, function(lhs, rhs)
+      return lhs.label < rhs.label
+    end)
+
+    table.sort(exporters_names, function(lhs, rhs)
+      return lhs < rhs
+    end)
+
+    return {
+      label = string.format('%s (%s)', label, table.concat(exporters_names, '/')),
+      key = key,
+      action = function()
+        return utils.menu(label .. ' via', commands, label .. ' via')
+      end,
+    }
+  end
+
+  local opts = {
+    submenu('h', 'Export to HTML file', 'html', {
+      emacs = {
+        command = 'org-html-export-to-html',
+      },
+      pandoc = {},
+    }),
+    submenu('l', 'Export to LaTex file', 'tex', {
+      emacs = {
+        command = 'org-latex-export-to-latex',
+      },
+      emacs_beamer = {
+        command = 'org-beamer-export-to-latex',
+      },
+      pandoc = {},
+    }),
+    submenu('p', 'Export to PDF file', 'pdf', {
+      emacs = {
+        command = 'org-latex-export-to-pdf',
+      },
+      emacs_beamer = {
+        command = 'org-beamer-export-to-pdf',
+      },
+      pandoc = {},
+    }),
+    submenu('m', 'Export to Markdown file', 'md', {
+      emacs = {
+        command = 'org-md-export-to-markdown',
+      },
+      pandoc = {
+        format = 'gfm',
+      },
+    }),
+    submenu('i', 'Export to iCalendar file', 'ics', {
```

Comment:
This shouldn't be a submenu since it has only 1 option.

Reply:
Done